### PR TITLE
fix: Fixed autopipeline performances.

### PR DESCRIPTION
**Pull request by ShogunPanda**
## What's in there?

This PR introduces new internal structures to compare the target nodes for different slots.
In particular, it assigns a numeric ID to each group of nodes (where by "group" I mean a master and its slaves).
This ID is then searchable using both the group name itself or by slot.

The change is needed to improve autopipeline performances, which degraded after the fix in #1224.

## Notes to reviewers

`playground.js` and `.gitignore` changes will be removed when the PR is ready for review/merge.

### CODE REVIEW

**Comment by mcollina** [comment]
Good for me!

File: lib/cluster/index.ts [hunk 1]
```diff
@@ -842,6 +849,24 @@ class Cluster extends EventEmitter {
           }
         }
 
+        // Assign to each node keys a numeric value to make autopipeline comparison faster.
+        this._groupsIds = {};
```
Comment (AVVS) [inline lib/cluster/index.ts:853]: Maybe Object.create(null) - should be slightly faster?

Comment (ShogunPanda) [reply lib/cluster/index.ts]: Done!

**Comment by ShogunPanda** [comment]
I have rebased this after latest master. Are we good to merge this?

**Review by luin — APPROVED** [review]

**Pull request merged**